### PR TITLE
Fix zero bytes file upload

### PR DESCRIPTION
**Pull request by ramonakira**
* Fix zero-byte uploads
* Handle zero bytes upload case

‌

Opened by @ramonakira on 2022-03-16 11:56:46

### CODE REVIEW

**Comment by ramonakira** [comment]
@ramonakira added reviewers on 2022-03-16 11:56:46

@bsimons
@wadevries

File: rest_framework_tus/views.py [hunk 1]
```diff
@@ -145,6 +146,14 @@ def create(self, request, *args, **kwargs):
         # Validate headers
         headers = self.validate_success_headers(headers)
 
+        # Upload length is zero bytes, so PATCH will not be called.
+        # Do the needed stuff here:
```
Comment (wadevries) [inline rest_framework_tus/views.py]: @wadevries commented on 2022-03-16 14:08:41

This comment can be improved; what is the ‘needed stuff’?

**Review by ThomasKalverda — APPROVED** [review]
@thomaskalverda approved on 2022-03-17 14:08:35

File: rest_framework_tus/views.py [hunk 2]
```diff
@@ -93,18 +93,19 @@ def create(self, request, *args, **kwargs):
...
-            return Response('Invalid "Upload-Length". Maximum value: {}.'.format(max_file_size),
-                            status=status.HTTP_413_REQUEST_ENTITY_TOO_LARGE)
 
-        # If upload_length is not given, we expect the defer header!
-        if not upload_length or upload_length < 0:
+        if upload_length is None:  # We want to allow 0
+            # If upload_length is not given, we expect the defer header!
             if getattr(request, constants.UPLOAD_DEFER_LENGTH_FIELD_NAME, -1) != 1:
                 return Response('Missing "{Upload-Defer-Length}" header.', status=status.HTTP_400_BAD_REQUEST)
+        else:
```
Comment (wadevries) [inline rest_framework_tus/views.py]: @wadevries commented on 2022-03-17 14:10:30

`elif upload_length > max_file_size:`

Comment (ramonakira) [reply rest_framework_tus/views.py]: @ramonakira commented on 2022-03-17 14:38:23

Ofcourse! done!

**Review by wadevries — APPROVED** [review]
@wadevries approved on 2022-03-18 13:56:49

**Pull request merged**